openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -460,11 +461,14 @@ def get_all_courses(cls, org=None, filter_=None):
         # created. For tests using CourseFactory, use emit_signals=True.
         course_overviews = CourseOverview.objects.all()
 
-        if org:
+        if orgs:
             # In rare cases, courses belonging to the same org may be accidentally assigned
             # an org code with a different casing (e.g., Harvardx as opposed to HarvardX).
-            # Case-insensitive exact matching allows us to deal with this kind of dirty data.
-            course_overviews = course_overviews.filter(org__iexact=org)
+            # Case-insensitive matching allows us to deal with this kind of dirty data.
+            # We need to escape the org names provided in case they contain regex special
+            # chars for some reason.
```

Comment:
You are right. We don't need this because validation on org names/course key creation would prevent special regex characters from ever being used.